Expecto.Tests/Main.fs

```diff
@@ -2,13 +2,42 @@ module Main
 
 open Expecto
 open Expecto.Logging
+open OpenTelemetry.Resources
+open OpenTelemetry
+open OpenTelemetry.Trace
+open System.Threading
+open System.Diagnostics
+
+let serviceName = "Expecto.Tests"
+
+let logger = Log.create serviceName
+
+
+let resourceBuilder =
+    ResourceBuilder
+        .CreateDefault()
+        .AddService(serviceName = serviceName)
+
 
-let logger = Log.create "Expecto.Tests"
 
 [<EntryPoint>]
 let main args =
+  let activitySource = new ActivitySource(serviceName)
+  use traceProvider =
+          Sdk
+              .CreateTracerProviderBuilder()
+              .AddSource(serviceName)
+              .SetResourceBuilder(resourceBuilder )
+              .AddOtlpExporter()
+              .Build()
+  let tracer = traceProvider.GetTracer(serviceName)
+  // use span = tracer.StartActiveSpan("Expecto.main")
+  use span = tracer.StartRootSpan("Expecto.main")
   let test =
     Impl.testFromThisAssembly()
     |> Option.orDefault (TestList ([], Normal))
     |> Test.shuffle "."
-  runTestsWithCLIArgs [NUnit_Summary "bin/Expecto.Tests.TestResults.xml"] args test
+  runTestsWithCLIArgs [NUnit_Summary "bin/Expecto.Tests.TestResults.xml"; CLIArguments.ActivitySource activitySource] args test
```

Comment:
Here a consumer would pass in the ActivitySource. Otherwise they'd have to add the dedicated Expecto one to the `.AddSource` call on line 29.